openedx/core/lib/block_structure/block_structure.py

```diff
@@ -602,11 +603,57 @@ def remove_block(self, usage_key, keep_descendants):
                 for parent in parents:
                     self._add_relation(parent, child)
 
-    def remove_block_if(self, removal_condition, keep_descendants=False, **kwargs):
+    def create_universal_filter(self):
+        """
+        Returns a filter function that always returns True for all blocks.
+        """
+        return lambda block_key: True
+
+    def create_removal_filter(self, removal_condition, keep_descendants=False):
+        """
+        Returns a filter function that automatically removes blocks that satisfy
+        the removal_condition.
+
+        Arguments:
+            removal_condition ((usage_key)->bool) - A function that
+                takes a block's usage key as input and returns whether
+                or not to remove that block from the block structure.
+
+            keep_descendants (bool) - See the description in
+                remove_block.
+        """
+        return partial(
+            self.remove_block_if,
+            removal_condition=removal_condition,
+            keep_descendants=keep_descendants,
+        )
+
+    def remove_block_if(self, block_key, removal_condition, keep_descendants=False):
+        """
+        Removes the given block if it satisfies the removal_condition.
+        Returns True if the block was retained, and False if the block
+        was removed.
```

Comment:
Good point.  But `block_exists_after_filter` isn't an accurate name either.  The function is conditionally removing a block - and hence the original name.  If we change its return value, can we easily update its callers to negate the return?

Alternatively, we can change the input `removal_condition` to instead be a `filter_condition` which specifies whether the block should be retained.  And then rename to something like `filter_or_remove`.  And then keep the existing return value.